src/main.rs

```diff
@@ -69,29 +73,74 @@ impl Client {
     // Method should return an error when a connection already exists.
     // The client should send a handshake to the server.
     fn open(&mut self, addr: &str, server: Server) -> CommsResult<()> {
-        todo!()
+        match self.connections.get(addr) {
+            None => {
+                self.connections
+                    .insert(addr.to_string(), Connection::Open(server));
+                match self.send(
+                    addr,
+                    Message {
+                        msg_type: MessageType::Handshake,
+                        load: self.ip.clone(),
+                    },
+                ) {
+                    Err(_) => Err(CommsError::ConnectionNotFound(addr.to_string())),
```

Comment:
To mapowanie jest niepotrzebne, a nawet szkodlie, bo przykrywa wszystkie błędy zwrócone z metody `send()`.